src/components/common/PropTypes.ts

```diff
@@ -0,0 +1,6 @@
+export interface DynamicListProps<TModel> {
+  addText: string | ((listItems: Array<TModel>) => string);
+  modelFactory: () => TModel;
+  updateModel: (value: Array<TModel>) => void;
+  eachRender: (updateModel: (value: TModel) => void) => JSX.Element;
```

Comment:
maybe "renderChild" or "renderEntry" or "renderListEntry"?

Reply:
updateModel => onItemUpdated?

Reply:
I think renderListItem and onListItemUpdated to keep consistent with the rest of it